src/llamafactory/hparams/model_args.py

```diff
@@ -239,6 +243,11 @@ def __post_init__(self):
         if self.new_special_tokens is not None:  # support multiple special tokens
             self.new_special_tokens = [token.strip() for token in self.new_special_tokens.split(",")]
 
+        if self.visual_inputs:
+            assert self.visual_inputs_type is not None, "Must specify visual inputs type while using visual input."
```

Comment:
try to move all visual_inputs_type to template

Reply:
Oaky

Reply:
There is a question that template is belong to `data_args`.However there is some place that I need `visual_inputs_type` to infer somethings eg.(how to frezze `vision_model`,how to load tokenizer/processer) and func like `load_model` should never get related to `data_args`.

Reply:
Let me think for a while

Reply:
In the case like `create_ref_model` I dont even have model to get `model.config.model_type`.If remove it from `model_args` I will need tons of change to core function which will influence a lots.

Reply:
Wait for our discussion🤗

Reply:
Any result?

Reply:
You can feel free to support the pairwise dataset and do enough tests, we will notify you in the near future🤗

Reply:
So i need to fix pairwise support for llava-like model also?I think I have some time to solve those.

Reply:
Finally, you might have to do that. For now, you can feel free to do or not do that.

Reply:
Oh any If you need any test log (after commit https://github.com/hiyouga/LLaMA-Factory/pull/4377/commits/3c2ecbab75e104256bfebe61e901a686e641901a) I'll send you.I think keeping test log is important for tracing.

Reply:
Please send your wechat to my email: muller.dreamer@qq.com, we can discuss more🤗